### PR TITLE
CORS-3265: capz: Add Accelerated Networking

### PR DESCRIPTION
**Pull request by rna-afk**
Adding the option to enable AcceleratedNetworking in control-plane.

### CODE REVIEW

File: pkg/asset/machines/azure/azuremachines.go [hunk 1]
```diff
@@ -133,6 +133,12 @@ func GenerateMachines(platform *azure.Platform, pool *types.MachinePool, userDat
 				AllocatePublicIP:       false,
 				EnableIPForwarding:     false,
 				SecurityProfile:        securityProfile,
+				NetworkInterfaces: []capz.NetworkInterface{
+					{
+						SubnetName:            subnet,
+						AcceleratedNetworking: additionalCapabilities.UltraSSDEnabled,
```
Comment (jhixson74) [inline pkg/asset/machines/azure/azuremachines.go:139]: I think this should just be set to true. SSD capabilities are for disks, not networking interfaces.

**Comment by sadasu** [comment]
/override  ci/prow/okd-images

Not expected to pass

**Pull request merged**